kpi_dashboard/models/kpi_kpi.py

```diff
@@ -84,6 +88,27 @@ def write(self, vals):
             vals["value_last_update"] = fields.Datetime.now()
         return super().write(vals)
 
+    def _get_code_input_dict(self):
+        return {
+            "self": self,
+            "model": self.browse(),
+        }
+
+    def _forbidden_code(self):
+        return "(commit|rollback)"
```

Comment:
Shouldnt you also forbid `execute` ?

Reply:
Well, in that case, it will avoid to have a compute value code that makes an SQL query, that can be interesting.

Reply:
Good Point, I have added execute to the forbidden words. I also added `getattr`, as you could do something like:
```
getattr(self.cr, 'exec'+'ute')('com'+'mit')
```

Reply:
But it's the contrary. @legalsylvain is saying that this prevent manual SQL queries, and the only problem I see is to commit that, so no need to forbid it.